src/modules/migrations/v090Migrations/migrateSenses.js

```diff
@@ -0,0 +1,34 @@
+export default function migrateSenses(actorData, updateData) {
+  const old = actorData?.system?.attributes?.senses;
+
+  const newSenses = Object.entries(old).reduce((acc, [sense, distance]) => {
+    const mileUnits = ['mile', 'miles', 'mi.', 'mi'];
+    const meterUnits = ['meter', 'meters', 'm.', 'm'];
+    const kilometerUnits = ['kilometer', 'kilometers', 'km.', 'km'];
+    const unlimitedUnits = ['unlimited', 'infinite'];
+
+    if (mileUnits.some((el) => distance.includes(el))) {
+      acc[sense] = { distance: Number(parseInt(distance, 10)), unit: 'miles' };
+      return acc;
+    }
+    if (kilometerUnits.some((el) => distance.includes(el))) {
+      acc[sense] = { distance: Number(parseInt(distance, 10)), unit: 'kilometers' };
+      return acc;
+    }
+    if (meterUnits.some((el) => distance.includes(el))) {
+      acc[sense] = { distance: Number(parseInt(distance, 10)), unit: 'meters' };
+      return acc;
+    }
```

Comment:
The same code is repeated almost exactly the same in each of these blocks, which is a good indication that we can cut down on duplication.

One option here is to define some objects which contain both the unit and the strings you're looking for. Something like this, maybe:
```js
unitMap = {
  miles: ['mile', 'miles', 'mi.', 'mi'],
  meters: ['meter', 'meters', 'm.', 'm'],
  kilometers: ['kilometer', 'kilometers', 'km.', 'km'],
  unlimited: ['unlimited', 'infinite']
}

const newSenses = Object.entries(old).reduce((acc, [sense, distance]) => {
  for (const [unit, strings] of Object.entries(unitMap)) {
    if (strings.some((el) => distance.includes(el))) {
      acc[sense] = { distance: Number(parseInt(distance, 10)), unit };
      return acc;
    }
  }

  acc[sense] = { distance: (Number(parseInt(distance, 10)) || 0), unit: 'feet' };

  return acc;
}, {});
```